RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,609 @@
+
+  std::vector<GlobalVector> splittedClusterDirections(
+      const reco::Candidate&,
+      const TrackerTopology* const,
+      const PixelClusterParameterEstimator*,
+      const reco::Vertex&,
+      int,
+      const edmNew::DetSetVector<SiPixelCluster>&);  //if not working,: args=2 auto
+
+  std::pair<double[jetDimX][jetDimY][Nover][Npar], double[jetDimX][jetDimY][Nover]> SeedEvaluation(
+      tensorflow::NamedTensorList, std::vector<std::string>);
+};
+
+DeepCoreSeedGenerator::DeepCoreSeedGenerator(const edm::ParameterSet& iConfig, const DeepCoreCache* cache)
+    : vertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      pixelClusters_(
+          consumes<edmNew::DetSetVector<SiPixelCluster>>(iConfig.getParameter<edm::InputTag>("pixelClusters"))),
+      cores_(consumes<edm::View<reco::Candidate>>(iConfig.getParameter<edm::InputTag>("cores"))),
+      topoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>()),
+      ptMin_(iConfig.getParameter<double>("ptMin")),
+      deltaR_(iConfig.getParameter<double>("deltaR")),
+      chargeFracMin_(iConfig.getParameter<double>("chargeFractionMin")),
+      centralMIPCharge_(iConfig.getParameter<double>("centralMIPCharge")),
+      pixelCPE_(iConfig.getParameter<std::string>("pixelCPE")),
+      weightfilename_(iConfig.getParameter<edm::FileInPath>("weightFile").fullPath()),
+      inputTensorName_(iConfig.getParameter<std::vector<std::string>>("inputTensorName")),
+      outputTensorName_(iConfig.getParameter<std::vector<std::string>>("outputTensorName")),
+      probThr(iConfig.getParameter<double>("probThr")),
+      session_(tensorflow::createSession(cache->graph_def))
+
+{
+  produces<TrajectorySeedCollection>();
+  produces<reco::TrackCollection>();
+}
+
+DeepCoreSeedGenerator::~DeepCoreSeedGenerator() {}
+
+void DeepCoreSeedGenerator::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  auto result = std::make_unique<TrajectorySeedCollection>();
+  auto resultTracks = std::make_unique<reco::TrackCollection>();
+
+  tensorflow::setLogging("3");
+  const tensorflow::TensorShape input_size_eta({1, 1});
+  const tensorflow::TensorShape input_size_pt({1, 1});
+  const tensorflow::TensorShape input_size_cluster({1, jetDimX, jetDimY, Nlayer});
+  std::vector<std::string> output_names;
+  output_names.push_back(outputTensorName_[0]);
+  output_names.push_back(outputTensorName_[1]);
+
+  using namespace edm;
+  using namespace reco;
+
+  iSetup.get<IdealMagneticFieldRecord>().get(magfield_);
+  iSetup.get<GlobalTrackingGeometryRecord>().get(geometry_);
+  iSetup.get<TrackingComponentsRecord>().get("AnalyticalPropagator", propagator_);
+
+  const auto& inputPixelClusters_ = iEvent.get(pixelClusters_);
+  allSiPixelClusters_.clear();
+  allSiPixelClusters_.reserve(
+      inputPixelClusters_.dataSize());  // this is important, otherwise push_back invalidates the iterators
```

Comment:
it looks like these are not used anywhere later, remove?

Reply:
removed